elastic/kibana
# D3 Security Sub Actions Connector

## Summary

This code comes from a PR that was [originally opened](https://github.com/elastic/kibana/pull/144955) by the team at D3 Security. In an effort to get this merged, I have branched off of that PR and we will go forward with this one. This PR takes D3's connector code, refactors to use sub actions, and adds the appropriate tests and documentation. 

## Testing
[Credentials](https://p.elstc.co/paste/fGEGYTAW#asf-vXL17gL4yqKOow3jtT5fvneJXmtrMtv4gn7xEYG)

1. Run kibana with this es server configured in [`kibana.dev.yml`](https://p.elstc.co/paste/ouvJvE0P#y4Bo9zKuHaxp4qllxIh2tgX9MAdEWko-dYOklYfYfP+). This is the server I have set up with the D3 instance already. There is some configuration you need to do in D3 to establish the connection, so using this server will allow you to skip that step. 
2. Create D3 Security connectors with above credentials
3. Test the connector by using the "Test" tab in connectors, or by running a rule that triggers a D3 Security action.
     <img width="1331" alt="Screenshot 2023-05-31 at 4 24 06 PM" src="https://github.com/elastic/kibana/assets/6935300/7544b393-c73a-43a5-b063-d9d8884382c8">
4. Confirm results in D3 interface. Should look like:
     <img width="1341" alt="Screenshot 2023-05-31 at 4 25 22 PM" src="https://github.com/elastic/kibana/assets/6935300/682d66e5-dd14-46c1-aefa-a2410af42d36">
     <img width="1343" alt="Screenshot 2023-05-31 at 4 25 38 PM" src="https://github.com/elastic/kibana/assets/6935300/4b13e967-2a0e-4123-a798-40ceb1db13a3">

## More info
Below is the [original PR](https://github.com/elastic/kibana/pull/144955) description:

> Introduces a new built-in action type and sends the ElasticSearch alerts to D3 Security SOAR for incident response, workflow automation and deep investigation. This integrated alert action allows you to create Events or trigger Playbook workflow actions automatically in the D3 Security SOAR platform.
> 
> The new action type and the action connector with D3 Security SOAR are as follows:
> 
> <img width="224" alt="D3ConnectorCard2" src="https://user-images.githubusercontent.com/109116834/200965642-6fef1a0b-232c-40c2-8556-d5abc50640f5.png">
> <img width="680" alt="Screenshot 2023-05-31 at 4 07 46 PM" src="https://github.com/elastic/kibana/assets/6935300/813b9c57-6e61-49e9-aed7-b6a21843280c">
> 
> 
> and action params form available in Create Alert form:
> 
> <img width="555" alt="D3_new-Form" src="https://user-images.githubusercontent.com/109116834/201801116-6b84fd2a-891c-4b88-af04-9d193efc7d0a.png">
